radsat-sk/src/main.c

```diff
@@ -7,104 +7,376 @@
+/** Communication Receive Task Priority. Constantly listening for messages; high priority task. */
+static const int communicationRxTaskPriority = configMAX_PRIORITIES - 2;
+
+/** Dosimeter Collection Task Priority. Periodically collects payload data; medium priority task. */
+static const int dosimeterCollectionTaskPriority = configMAX_PRIORITIES - 3;
+/** Image Capture Task Priority. Periodically collects image data; medium priority task. */
+static const int imageCaptureTaskPriority = configMAX_PRIORITIES - 3;
+/** ADCS Capture Task Priority. Periodically collects ADCS data; medium priority task. */
+static const int adcsCaptureTaskPriority = configMAX_PRIORITIES - 3;
+
+/** Telemetry Collection Task Priority. Periodically collects satellite telemetry; low priority task. */
+static const int telemetryCollectionTaskPriority = configMAX_PRIORITIES - 4;
+/** Satellite Watchdog Task Priority. Routinely pets (resets) satellite subsystem watchdogs; low priority task. */
+static const int satelliteWatchdogTaskPriority = configMAX_PRIORITIES - 4;
+/** Mission Init Task Priority. Does initializations that need to be ran post-scheduler; low priority task. */
+static const int missionInitTaskPriority = configMAX_PRIORITIES - 4;
+
+
+/***************************************************************************************************
+                                       PRIVATE FUNCTION STUBS
+***************************************************************************************************/
+
+static void initBoard(void);
+static int initDrivers(void);
+static int initTime(void);
+static int initSubsystems(void);
+
+static int initObcWatchdog(void);
+static int initMissionTasks(void);
+
+void MissionInitTask(void* parameters);
+
+
+/***************************************************************************************************
+                                                MAIN
+***************************************************************************************************/
+
+/**
+ * The main application entry point.
+ * @return Never returns.
+ */
+int main(void) {
+
+	// initialize internal OBC board settings
+	initBoard();
+
+	// initialize the Hardware Abstraction Library (HAL) drivers
+	initDrivers();
+
+	// initialize external components and the Satellite Subsystem Interface (SSI)
+	initSubsystems();
+
+	// initialize the internal OBC watchdog, and start a task that automatically pets it
+	initObcWatchdog();
+
 #ifdef TEST
-#include <RTestSuite.h>
-#endif /* TEST */
-
-#define ENABLE_MAIN_TRACES 1
-#if ENABLE_MAIN_TRACES
-	#define MAIN_TRACE_INFO			TRACE_INFO
-	#define MAIN_TRACE_DEBUG		TRACE_DEBUG
-	#define MAIN_TRACE_WARNING		TRACE_WARNING
-	#define MAIN_TRACE_ERROR		TRACE_ERROR
-	#define MAIN_TRACE_FATAL		TRACE_FATAL
-#else
-	#define MAIN_TRACE_INFO(...)	{ }
-	#define MAIN_TRACE_DEBUG(...)	{ }
-	#define MAIN_TRACE_WARNING(...)	{ }
-	#define MAIN_TRACE_ERROR		TRACE_ERROR
-	#define MAIN_TRACE_FATAL		TRACE_FATAL
-#endif
 
```

Comment:
Would like to confirm that the `#ifdef TEST` is where you plan on calling your RTestSuite function

Reply:
For now yeah... testing needs a lot more thought and consideration though